[mongoose] Can't use 'new' with extended models (discriminators)

- [x] I tried using the latest `mongoose/mongoose.d.ts` file in this repo and had problems.
- [ ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] I want to talk about `mongoose/mongoose.d.ts`.
  - The authors of that type definition are cc/ @simonxca @horiuchi 


**_example code:_**
```ts
...
const base: Model<ibase> = model<ibase>('testfour', baseSchema)
const extended = base.discriminator('extendedS', extendedSchema)
const x = new extended({
  username: 'hi',     // required in baseSchema
  email: 'beddiw',    // required in extededSchema
})
...
```

_**error:**_
```
error TS2351: Cannot use 'new' with an expression whose type lacks a call or construct signature
```

_**using:**_
```
tsc: v2.0.0
@types/mongoose v4.5.33",
mongoose v4.6.0"
```

--- a/mongoose/mongoose-tests.ts
+++ b/mongoose/mongoose-tests.ts
@@ -1124,7 +1124,6 @@ MongoModel.create([{ type: 'jelly bean' }, {
   arg[0].save();
   arg[1].save();
 });
-MongoModel.discriminator('M', new mongoose.Schema({name: String}));
 MongoModel.distinct('url', { clicks: {$gt: 100}}, function (err, result) {
 });
 MongoModel.distinct('url').exec(cb);
@@ -1387,3 +1386,15 @@ Final2.staticProp;
 var final2 = new Final2();
 final2.prop;
 final2.method;
+interface ibase extends mongoose.Document {
+  username: string;
+}
+interface extended extends ibase {
+  email: string;
+}
+const base: mongoose.Model<ibase> = mongoose.model<ibase>('testfour', null)
+const extended: mongoose.Model<extended> = base.discriminator<extended>('extendedS', null);
+const x = new extended({
+  username: 'hi',     // required in baseSchema
+  email: 'beddiw',    // required in extededSchema
+})

--- a/mongoose/mongoose.d.ts
+++ b/mongoose/mongoose.d.ts
@@ -2119,7 +2119,7 @@ declare module "mongoose" {
      * @param name discriminator model name
      * @param schema discriminator model schema
      */
-    discriminator(name: string, schema: Schema): T;
+    discriminator<U extends Document>(name: string, schema: Schema): Model<U>;
 
     /** Creates a Query for a distinct operation. Passing a callback immediately executes the query. */
     distinct(field: string, callback?: (err: any, res: any[]) => void): Query<any[]>;

